api/application.go

```diff
@@ -616,8 +602,13 @@ func (h ApplicationHandler) FactGet(ctx *gin.Context) {
 		_ = ctx.Error(result.Error)
 		return
 	}
-	key := ctx.Param(Key)
-	source := ctx.Param(Source)[1:]
+
+	source, key := qualifiedFact(ctx.Param(Key))
+	if key == "" {
+		h.FactList(ctx, source)
+		return
+	}
+
 	list := []model.Fact{}
 	result = h.DB(ctx).Find(&list, "ApplicationID = ? AND Key = ? AND source = ?", id, key, source)
```

Comment:
This is fine but generally we have been doing this in other places to reduce the raw sql.
```
db = h.DB(ctx)
db = db.Where("ApplicationID", id)
db = db.Where("Source", source)
db = db.Where("Key", key)
```